### PR TITLE
'Parkrun every date in the year' challenge

### LINKED ISSUE #98: parkrun every date of the year
(bit like stopwatch bingo but for each date of the year, inc 29th Feb - next Sat 29th Feb is in 2020)

Need to account for parkrun vs junior parkrun events - allow only Saturdays?

Record the year for each date achieved?
Provide 'next available Saturday' for each date that hasn't yet been achieved.

Prob requires #97 to be completed first as this will be a loooooong table of dates.

### PR DESCRIPTION
**Pull request by rboyatt**
A first attempt to close #98 - a new challenge for parkrun every date in the year.  This could probably do with some further refinement but thought I'd create a pull request so it doesn't get lost in my git branches (where it's already been sitting for ~2 months!).

image

### CODE REVIEW

**Comment by rboyatt** [comment]
Thanks to #130 and #131, there's alternative way to implement this now....except we'd lose the 'Possible on...' aspect.  If you think this is worth pursuing as a challenge, I'll look at redoing this given the new `challenge_on_dates` function.

**Comment by fraz3alpha** [comment]
@rboyatt , I think this challenge is going to have to be a special case. The fact that displaying it is going to be pages and pages of output for 366 dates makes this one need a bit more intelligence and its own function - so I think keeping it separate from the `on_dates` function is sensible. I'll have some more time over Christmas, so hopefully I'll have change to take another look at ways to improve the ways we display challenges so things like this work well.
TLDR; leave your code as-is for now I think.

**Comment by rboyatt** [comment]
Branch brought up to date with master to check if it still works (it does)

**Comment by rboyatt** [comment]
Given changes in #357 could potentially pursue this but in a collapsed by default state.  Any thoughts on whether it's worth including?  If not, no objections to just closing this request.

**Comment by rboyatt** [comment]
Given this has been open for almost 5 years, and aligned with position on new developments - closing this pull request.
